openpype/hosts/maya/tools/mayalookassigner/commands.py

```diff
@@ -107,6 +108,7 @@ def create_asset_id_hash(nodes):
     """
     node_id_hash = defaultdict(list)
     for node in nodes:
+        shapes = cmds.listRelatives(node, shapes=True, fullPath=True)
```

Comment:
Why do we need the listRelatives call - the VRayProxy node I believe is also a Shape node right? Why did it work for that without requiring to go down to extra child nodes?

I believe however you're calling this `create_asset_id_hash` the nodes you're passing should already include the shape nodes too. You definitely don't want to be getting shapes in a for loop like this over what initially already could be a massive nodes list.

Reply:
`create_asset_id_hashes` is being used on the selection [here](https://github.com/ynput/OpenPype/blob/3da1f72f96ff8c633667211e5d0294a41a45a93d/openpype/hosts/maya/tools/mayalookassigner/widgets.py#L118-L123). Where do you see it including the shapes?

Reply:
The `dag=True` flag includes the full hierarchy of all dag nodes. When selection is True there then it takes the one from commands that I passed above which explicitly takes the selection and its children. Both if and else branch there should include hierarchy downwards.

If they wouldn't that's a bug.

Also that, `cmds.ls` flag might need to also take `allPaths=True` if it requires all instances to be taken into account as well.

Reply:
Ok, think I've found the issue. This [method](https://github.com/ynput/OpenPype/blob/3da1f72f96ff8c633667211e5d0294a41a45a93d/openpype/hosts/maya/tools/mayalookassigner/commands.py#L78-L98) only lists the nodes added to the container and not its children which makes sense. Problem is that when loading in the aiStandIn, we only add the transform.
Question is whether we should add the shape or accommodate otherwise?

Reply:
Interesting. 

This is a tricky situation. The container object set should include all the nodes.

I believe at some point in time shapes got excluded from being in object sets die to issues with shader assignments with instances if the shape node is in an objectSet. (This was for referenced objects actually)

Then at some point the referenced nodes in containers were replaced with just adding the reference node to the containers instead of all its nodes.

But I assume that the loader for e.g. aiStandin still excludes the shape from the container objectSet which is a bit counter intuitive because the shape IS actually part of that loaded data to begin with.

The best solution now would be that whenever we're retrieving the members from the container, in particular in the look assigner that we do a single `shapes = cmds.listRelatives(container_members, shapes=True, fullPath=True, noIntermediate=True)` then we just add that to the result of that function. We'd still need to ensure it's unique so `list(set(members))` or alike after that.

Then let's add a clear comment about what we are solving that way, e.g. **Some containers, like loaded aiStandin don't include the shape directly in the container's objectSet so we need to include the shapes here ourselves**

What do you think?

Reply:
I think maybe I should just add the shape to the container node. The shape is not needed for the vray implementation, so its most likely that I've misunderstood something like doing the [ASS overhaul](https://github.com/ynput/OpenPype/pull/4449).

Reply:
It would make this logic incompatible however with any file that was loaded since that overhaul. I think we should be fine with including the shapes for now as I mentioned. We are triggering the call only once, shouldn't be too problematic.